### PR TITLE
#49 add more metrics to application-source

### PR DESCRIPTION
**Pull request by Zhanshaoxiong**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/master/ApplicationSource.scala [hunk 1]
```diff
@@ -38,4 +38,26 @@ private[master] class ApplicationSource(val application: ApplicationInfo) extend
...
+    override def getValue: Int = application.desc.maxCores.getOrElse(0)
+  })
+
+  metricRegistry.register(MetricRegistry.name("instances"), new Gauge[Int] {
+    val maxCores: Int = application.desc.maxCores.getOrElse(0)
+    val coresPerExecutor: Int = application.desc.coresPerExecutor.getOrElse(1)
+    override def getValue: Int = maxCores / coresPerExecutor
+  })
+
+  metricRegistry.register(MetricRegistry.name("requestExecutors"), new Gauge[Int] {
```
Comment (jiaoew1991) [inline core/src/main/scala/org/apache/spark/deploy/master/ApplicationSource.scala]: 命名统一一下

Comment (Zhanshaoxiong) [reply core/src/main/scala/org/apache/spark/deploy/master/ApplicationSource.scala]: done with 'request_executors'

**Comment by zheniantoushipashi** [comment]
retest this please

**Comment by zheniantoushipashi** [comment]
retest this please

**Comment by zheniantoushipashi** [comment]
retest this please

**Comment by zheniantoushipashi** [comment]
retest this please

**Comment by zheniantoushipashi** [comment]
retest this please

**Comment by zheniantoushipashi** [comment]
retest this please

**Comment by zheniantoushipashi** [comment]
retest this please

**Comment by zheniantoushipashi** [comment]
retest this please

**Comment by zheniantoushipashi** [comment]
retest this please

**Comment by zheniantoushipashi** [comment]
retest this please

**Comment by zheniantoushipashi** [comment]
retest this please

**Comment by zheniantoushipashi** [comment]
retest this please

**Comment by zheniantoushipashi** [comment]
retest this please

**Comment by zheniantoushipashi** [comment]
这个得改下  spark 各个模块的版本，可以参考下之前的 pr，

File: pom.xml [hunk 2]
```diff
@@ -118,7 +118,7 @@
     <sbt.project.name>spark</sbt.project.name>
     <slf4j.version>1.7.16</slf4j.version>
     <log4j.version>1.2.17</log4j.version>
-    <hadoop.version>2.6.5</hadoop.version>
+    <hadoop.version>2.7.1</hadoop.version>
```
Comment (zheniantoushipashi) [inline pom.xml:121]: 这个比较改动比较大，要对下

Comment (Zhanshaoxiong) [reply pom.xml:121]: org.apache.hadoop.mapreduce.lib.output.FileOutputCommitter
这个改动主要是希望使用到mapreduce.fileoutputcommitter.algorithm.version=2版本对于commitJob的调整，当mapreduce.fileoutputcommitter.algorithm.version=2时，在job commit时不再进行rename操作.
这篇文章有介绍: https://www.jianshu.com/p/0f523e54f9dc

Comment (zheniantoushipashi) [reply pom.xml:121]: 这里可以考虑把 hadoop-mapreduce-client-core 单独换成 2.7.1  或者单独打一个  2.6.5 的包，把相关改动 rebase 过来，ut 现在跑不过。

Comment (zheniantoushipashi) [reply pom.xml:121]: 这里可以考虑把 hadoop-mapreduce-client-core 单独换成 2.7.1 或者单独打一个 2.6.5 的包，把相关改动 rebase 过来，ut 现在跑不过。

**Comment by Zhanshaoxiong** [comment]
retest this please

**Pull request merged**